feat: adds filetypes to dap-view buffers

catgoose opened the pull request:
Adds filetypes to dap terminal buffer and dap-view action selector buffer

tlj commented:
This is great! I would love to be able to close the dap-view with 'q' through an autocmd keymap 👍

igorlfs reviewed (commented):
Hello!

Can you also add a note on the README, mentioning that users can now override filtetype-specific options using this new feature?

catgoose commented:
I updated readme with a section for the filetypes with example autocommands.  I also added a TOC to the readme, but if you don't like that I can remove it.

igorlfs commented:
> I updated readme with a section for the filetypes with example autocommands.

Thanks! The autocmd for the terminal stuff may be superseded by #6. If you're in a rush for this to be merged, we can remove it for now, but otherwise I think it's best to wait for the other PR.

> I also added a TOC to the readme

That's really nice, IMO!

catgoose commented:
> > I updated readme with a section for the filetypes with example autocommands.
> 
> Thanks! The autocmd for the terminal stuff may be superseded by #6. If you're in a rush for this to be merged, we can remove it for now, but otherwise I think it's best to wait for the other PR.
> 
> > I also added a TOC to the readme
> 
> That's really nice, IMO!

Yeah no rush, I'm closing the terminal by the autocmd in the meantime.

catgoose commented:
> > I updated readme with a section for the filetypes with example autocommands.
> 
> Thanks! The autocmd for the terminal stuff may be superseded by #6. If you're in a rush for this to be merged, we can remove it for now, but otherwise I think it's best to wait for the other PR.
> 
> > I also added a TOC to the readme
> 
> That's really nice, IMO!

I removed the terminal close autocommand if you want to merge this to address #13

igorlfs commented on lua/dap-view/term/options.lua:
```diff
@@ -12,6 +12,10 @@ M.set_options = function()
+
+    local buf = vim.bo[state.term_bufnr]
+    buf.filetype = "dap-view-term"
+    vim.api.nvim_set_option_value("buftype", "nofile", { scope = "local", buf = state.term_bufnr })
```
```suggestion
    buf.buftype = "nofile"
```

igorlfs commented on README.md:
```diff
@@ -163,6 +178,33 @@ NvimDapViewBreakpointSeparator
+These filetypes can be used to override buffer and window options set by `nvim-dap-view`
+
+<details>
+    <summary>Example autocommands</summary>
```
```suggestion
    <summary>Example autocommand</summary>
```

igorlfs commented on README.md:
````diff
@@ -163,6 +178,33 @@ NvimDapViewBreakpointSeparator
+
+```lua
+vim.api.nvim_create_autocmd({ "FileType" }, {
+  pattern = { "dap-view", "dap-view-term", "dap-repl" }, -- dap-repl is set by `dap`
````
```suggestion
  pattern = { "dap-view", "dap-view-term", "dap-repl" }, -- dap-repl is set by `nvim-dap`
```

igorlfs commented on README.md:
```diff
@@ -163,6 +178,33 @@ NvimDapViewBreakpointSeparator
+vim.api.nvim_create_autocmd({ "FileType" }, {
+  pattern = { "dap-view", "dap-view-term", "dap-repl" }, -- dap-repl is set by `dap`
+  callback = function(evt)
+    vim.api.nvim_buf_set_keymap(evt.buf, "n", "q", "<cmd>q<CR>", { noremap = true, silent = true })
+  end,
```
```suggestion
    vim.keymap.set("n", "q", "<C-w>q", { silent = true, buffer = evt.buf })
  end,
```

igorlfs reviewed (commented):
Left some minor suggestions. Can you squash `c1ed5d6` into `5a46e7e`?

catgoose commented:
Alrighty, pushed a commit.  Let me know if you need me to change anything else!

igorlfs commented:
Good to go!
